syntax = "proto3";

import "enums.proto";

message Empty {}

message JobState {
  uint32 job_id = 1;
  JobStatus status = 2; 
}

message JobDescription {
    uint64 job_id = 1;
    string job_type = 2;
    string command = 3;
    string working_directory = 4;
    bool needs_data_dir = 5;
    string num_steps_arg = 6;
    uint64 num_steps = 7;
    bool has_duration = 8;
    uint64 duration = 9;
    string mode = 10;
    uint64 mps_thread_percentage = 11;
    uint64 scale_factor = 12;
}
